test/DeblockTest.cc

```diff
@@ -0,0 +1,472 @@
+    const uint8_t *_thresh1);
+void aom_lpf_vertical_4_dual_sse2(uint8_t *s, int32_t p, const uint8_t *blimit0,
+                                  const uint8_t *limit0, const uint8_t *thresh0,
+                                  const uint8_t *blimit1, const uint8_t *limit1,
+                                  const uint8_t *thresh1);
+void aom_lpf_horizontal_8_dual_sse2(
+    uint8_t *s, int32_t p, const uint8_t *_blimit0, const uint8_t *_limit0,
+    const uint8_t *_thresh0, const uint8_t *_blimit1, const uint8_t *_limit1,
+    const uint8_t *_thresh1);
+}
+
+// define the common params
+#define LOOP_PARAM \
+    int p, const uint8_t *blimit, const uint8_t *limit, const uint8_t *thresh
+#define DUAL_LOOP_PARAM                                                        \
+    int p, const uint8_t *blimit0, const uint8_t *limit0,                      \
+        const uint8_t *thresh0, const uint8_t *blimit1, const uint8_t *limit1, \
+        const uint8_t *thresh1
+
+// typedef the function type and test param
+using LbdLoopFilterFunc = void (*)(uint8_t *s, LOOP_PARAM);
+using LbdDualLoopFilterFunc = void (*)(uint8_t *s, DUAL_LOOP_PARAM);
+using HbdLoopFilterFunc = void (*)(uint16_t *s, LOOP_PARAM, int bd);
+using HbdDualLoopFilterFunc = void (*)(uint16_t *s, DUAL_LOOP_PARAM, int bd);
+
+using HbdLpfTestParam =
+    ::testing::tuple<HbdLoopFilterFunc, HbdLoopFilterFunc, int>;
+using HbdDualLpfTestParam =
+    ::testing::tuple<HbdDualLoopFilterFunc, HbdDualLoopFilterFunc, int>;
+using LdbLpfTestParam =
+    ::testing::tuple<LbdLoopFilterFunc, LbdLoopFilterFunc, int>;
+using LdbDualLpfTestParam =
+    ::testing::tuple<LbdDualLoopFilterFunc, LbdDualLoopFilterFunc, int>;
+
+uint8_t get_outer_thresh(ACMRandom *rnd) {
+    return static_cast<uint8_t>(rnd->PseudoUniform(3 * MAX_LOOP_FILTER + 5));
+}
+
+uint8_t get_inner_thresh(ACMRandom *rnd) {
+    return static_cast<uint8_t>(rnd->PseudoUniform(MAX_LOOP_FILTER + 1));
+}
+
+uint8_t get_hev_thresh(ACMRandom *rnd) {
+    return static_cast<uint8_t>(rnd->PseudoUniform(MAX_LOOP_FILTER + 1) >> 4);
+}
+
+/**
+ * @brief Unit test for deblocking assembly functions
+ *
+ * Test strategy:
+ * Feed src data generated randomly and all possible input,
+ * then check the dst buffer from target function and reference
+ * function.
+ *
+ * Expect result:
+ * The src buffer modified by deblocking from targeted function
+ * should be identical with the values from reference function.
+ *
+ * Test coverage:
+ * Test cases:
+ * blimit: [0, 3 * MAX_LOOP_FILTER + 4) as per spec 7.14
+ * limit: [0, MAX_LOOP_FILTER]
+ * hevThresh: [0, MAX_LOOP_FILTER] >> 4
+ * bitdepth: 8, 10, 12
+ *
+ */
+template <typename Sample, typename FuncType, typename TestParamType>
+class LoopFilterTest : public ::testing::TestWithParam<TestParamType> {
+  public:
+    enum LpfType { SINGLE, DUAL };
+    virtual ~LoopFilterTest() {
+    }
+
+    void SetUp() override {
+        lpf_tst_ = ::testing::get<0>(this->GetParam());
```

Comment:
please re-place it with TEST_GET_PARAM

Reply:
Could not replace it with TEST_GET_PARAM, compilation failed since the definitions are different.